pkg/apis/core/types.go

```diff
@@ -3503,6 +3507,10 @@ type EphemeralContainerCommon struct {
 	// +featureGate=InPlacePodVerticalScaling
 	// +optional
 	ResizePolicy []ContainerResizePolicy
+	// Restart policy for the container.
+	// +featureGate=SidecarContainers
+	// +optional
+	RestartPolicy *RestartPolicy
```

Comment:
Why do we need this in Ephemeral?  It's not actually usable here, right?

Reply:
https://github.com/kubernetes/kubernetes/blob/45b96eae989efef9380fadc1e46957dc6174f351/pkg/apis/core/types.go#L3546-L3548

Sadly, that is because of this.

Reply:
yes, this is inevitable with the current types

Reply:
How can we document "this cannot be used in ephemeral containers"

Reply:
added a comment to ephemeralContainerCommon's restartPolicy field.
For now, this seems best.